src/pages/Admin/MapPreview/MapPreview.tsx

```diff
@@ -103,7 +124,7 @@ const MapPreview: React.FC<MapPreviewProps> = ({
 
         {mapBackground &&
           !editing &&
-          rooms?.map((room) => (
+          mapRooms?.map((room) => (
```

Comment:
higher in the code you have `mapRooms!`, which is bug-prone. Has something changed here that you use `?`?